### PR TITLE
UCT/UD: Fix assert compilation issue with O0

### LINKED ISSUE #1859: compilation error with gcc 4.8.5, only at level -O0 (oh-zero)
CentOS 7, system gcc 4.8.5.  Compiles with -O1 and up.  UCX master @ cf49bb3609b2dc77a75da4bf16b2fe697e806d26

```
make[2]: Entering directory `/home/tc/src/ucx/git/build/src/uct'
  CC       base/libuct_la-uct_md.lo
...
  CC       ib/ud/base/libuct_la-ud_iface.lo
In file included from /home/tc/src/ucx/git/build/../source/src/ucs/datastruct/list.h:10:0,
                 from /home/tc/src/ucx/git/build/../source/src/uct/base/uct_worker.h:12,
                 from ../../../source/src/uct/ib/ud/base/ud_iface.h:11,
                 from ../../../source/src/uct/ib/ud/base/ud_iface.c:7:
../../../source/src/uct/ib/ud/base/ud_iface.c: In function ‘uct_ud_iface_calc_gid_len’:
../../../source/src/uct/ib/ud/base/ud_iface.c:392:63: error: case label does not reduce to an integer constant
     UCS_STATIC_ASSERT((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
                                                               ^
/home/tc/src/ucx/git/build/../source/src/ucs/sys/compiler.h:31:30: note: in definition of macro ‘UCS_STATIC_ASSERT’
      switch(0) {case 0:case (_cond):;}
                              ^
../../../source/src/uct/ib/ud/base/ud_iface.c:395:56: error: case label does not reduce to an integer constant
     UCS_STATIC_ASSERT(UCT_IB_GRH_LEN - offsetof(struct ibv_grh, dgid) == ipv6_len);
                                                        ^
/home/tc/src/ucx/git/build/../source/src/ucs/sys/compiler.h:31:30: note: in definition of macro ‘UCS_STATIC_ASSERT’
      switch(0) {case 0:case (_cond):;}
                              ^
make[2]: *** [ib/ud/base/libuct_la-ud_iface.lo] Error 1
make[2]: Leaving directory `/home/tc/src/ucx/git/build/src/uct'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/tc/src/ucx/git/build'
make: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by brminich**
fixes #1859

### CODE REVIEW

**Comment by shamisp** [comment]
@yosefe why we didn't catch this with regression ?

File: src/uct/ib/ud/base/ud_iface.c [hunk 1]
```diff
@@ -389,10 +389,10 @@ static void uct_ud_iface_calc_gid_len(uct_ud_iface_t *iface)
     uint16_t *local_gid_u16 = (uint16_t*)iface->super.gid.raw;
 
     /* Make sure that daddr in IPv4 resides in the last 4 bytes in GRH */
-    UCS_STATIC_ASSERT((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
+    ucs_assert_always((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_iface.c]: pls use #define UCT_UD_IPV4_LEN to use static assert - rather that reverting to runtime assert

**Comment by brminich** [comment]
probably also need to add build with --enable-debug to Jenkins

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4645/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4646/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4649/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@yosefe let's add debug build on jenkins asap.

**Comment by hoopoepg** [comment]
it should be there already

**Comment by shamisp** [comment]
then why we didn't step on it.

**Comment by brminich** [comment]
@shamisp, it's added as a part of this commit

**Comment by shamisp** [comment]
@brminich missed it. thx!

**Pull request merged**